pkg/controller/config/match/match.go

```diff
@@ -0,0 +1,89 @@
+package match
+
+import (
+	"sync"
+
+	configv1alpha1 "github.com/open-policy-agent/gatekeeper/apis/config/v1alpha1"
+)
+
+type Process string
+
+const (
+	Audit   = Process("audit")
+	Sync    = Process("sync")
+	Webhook = Process("webhook")
+	Star    = Process("*")
+)
+
+type Set struct {
+	Mux                sync.RWMutex
+	excludedNamespaces map[Process]map[string]bool
+}
+
+var allProcesses = []Process{
+	Audit,
+	Webhook,
+	Sync,
+}
+
+var configMapSet = &Set{
+	excludedNamespaces: make(map[Process]map[string]bool),
+}
+
+func GetSet() *Set {
+	return configMapSet
+}
+
+func newSet() *Set {
+	return &Set{
+		excludedNamespaces: make(map[Process]map[string]bool),
+	}
+}
+
+func (s *Set) update(entry []configv1alpha1.MatchEntry) {
+	s.Mux.RLock()
```

Comment:
I think this should be Lock() for read and write.

Reply:
thanks, fixed